src/CaptureFile/CaptureFile.cpp

```diff
@@ -55,23 +86,39 @@ ErrorMessageOr<void> CaptureFileImpl::Initialize() {
 
   fd_ = std::move(fd_or_error.value());
 
-  OUTCOME_TRY(ReadHeaderAndSectionList());
+  OUTCOME_TRY(ReadHeader());
+  OUTCOME_TRY(ReadSectionList());
+  OUTCOME_TRY(CalculateCaptureSectionSize());
+
   return outcome::success();
 }
 
-ErrorMessageOr<void> CaptureFileImpl::ReadHeaderAndSectionList() {
-  OUTCOME_TRY(header, orbit_base::ReadFullyAtOffset<CaptureFileHeader>(fd_, 0));
-  header_ = header;
-
-  if (std::memcmp(header_.signature.data(), kFileSignature, header_.signature.size()) != 0) {
-    return ErrorMessage{"Invalid file signature"};
+ErrorMessageOr<void> CaptureFileImpl::CalculateCaptureSectionSize() {
+  // Calculate capture section size
+  if (header_.section_list_offset == 0) {
```

Comment:
This means that there are no additional sections, right?

Reply:
yes. added a comment.